부산대_BE_진서현_3주차_3단계_re

jinseohyun1228 opened the pull request:
###구현 목록
각 화면에 페이지 기능을 추가하고, 헤더를 푸가했습니다.

✔️ 관리자용 상품페이지에 물건을 페이지로 볼 수 있도록 구현
✔️ 일반 상품페이지에 물건을 페이지로 볼 수 있도록 구현
✔️ 위시리스트 페이지에 페이지로 볼 수 있도록 구현

-------
빠르게 2단계 PR 리뷰 확인하고 반영하겠습니다..!!
이번주는 너무 진도가 늦었던 것 같습니다! 죄송합니다!

wotjd243 committed:
Initial commit

wotjd243 committed:
Initial commit

wotjd243 committed:
feat: set up the project

jinseohyun1228 committed:
Docs : 파일 형식자 변경

jinseohyun1228 committed:
Test : 컨트롤러 잘 돌아가는지..ㅎ

jinseohyun1228 committed:
Feat: 컨트롤러 작성

jinseohyun1228 committed:
Feat: 컨버터 작성

jinseohyun1228 committed:
Feat: Product 엔티티

jinseohyun1228 committed:
Feat: dto 작성

jinseohyun1228 committed:
Feat: 컨트롤러 작성

jinseohyun1228 committed:
fix : getter 없던 오류 추가

jinseohyun1228 committed:
feat: 요청 객체 추가

jinseohyun1228 committed:
fix : 1. 메서드 접근 제어자 수정, 2. id 발급 전략 선택 3.

jinseohyun1228 committed:
fix : 1. 메서드 접근 제어자 수정, 2. id 발급 전략 선택 3. 쿼리파라미터 타입형 long으로 수정 4. 불필요한 예외처리 삭제

jinseohyun1228 committed:
Feat: thymeleaf 공부

jinseohyun1228 committed:
style : ProductRequest의 줄바꿈

jinseohyun1228 committed:
Feat: Product에 toString 추가

jinseohyun1228 committed:
Feat: 타임리프를 적용한 html 문서 추가

jinseohyun1228 committed:
Feat:수정을 위한 별도의 페이지 추가

jinseohyun1228 committed:
Feat: 관리자페이지 관련 컨트롤러 추가

jinseohyun1228 committed:
Feat: 폼에서 put 메서드를 쓰기 위한 빈등록

jinseohyun1228 committed:
chore : 타임리프 사용을 위한 의존성 추가

jinseohyun1228 committed:
Style : 페이지 한글로 통일

jinseohyun1228 committed:
Style : 페이지 한글로 통일

jinseohyun1228 committed:
refactor : 전체 메서드의 매핑 url에 처음을  rest로 수정

jinseohyun1228 committed:
refactor : @RestMapping 사용하고, 변수명 바꿈(오타수정등)

jinseohyun1228 committed:
refactor :쿼리 파라미터를 @ModelAttribute 사용해서 한버ㅓㄴ에 받도록

jinseohyun1228 committed:
style: 필요없는 거 지움

jinseohyun1228 committed:
chore : application.properties 에 jdbc 설정 추가

jinseohyun1228 committed:
style : toString 삭제

jinseohyun1228 committed:
feat : JDBC를 사용한 레파지토리 Dao 추가

jinseohyun1228 committed:
feat : JDBC Dao 를 사용해서 DB 기능 추가

jinseohyun1228 committed:
style : MapToProductTransformer -> ProductTransformer클래스 명 변경

jinseohyun1228 committed:
docs : README.md 작성

wotjd243 committed:
feat: set up the project

jinseohyun1228 committed:
Merge remotes/a-repo/step3 into step0

jinseohyun1228 committed:
부산대_BE_진서현_2주차_과제 _Step0 (#60)

* Initial commit

* feat: set up the project

* Docs : 파일 형식자 변경

* Test : 컨트롤러 잘 돌아가는지..ㅎ

* Feat: 컨트롤러 작성

* Feat: 컨버터 작성

* Feat: Product 엔티티

* Feat: dto 작성

* Feat: 컨트롤러 작성

* fix : getter 없던 오류 추가

* feat: 요청 객체 추가

* fix : 1. 메서드 접근 제어자 수정, 2. id 발급 전략 선택 3.

* fix : 1. 메서드 접근 제어자 수정, 2. id 발급 전략 선택 3. 쿼리파라미터 타입형 long으로 수정 4. 불필요한 예외처리 삭제

* Feat: thymeleaf 공부

* style : ProductRequest의 줄바꿈

* Feat: Product에 toString 추가

* Feat: 타임리프를 적용한 html 문서 추가

* Feat:수정을 위한 별도의 페이지 추가

* Feat: 관리자페이지 관련 컨트롤러 추가

* Feat: 폼에서 put 메서드를 쓰기 위한 빈등록

* chore : 타임리프 사용을 위한 의존성 추가

* Style : 페이지 한글로 통일

* Style : 페이지 한글로 통일

* refactor : 전체 메서드의 매핑 url에 처음을  rest로 수정

* refactor : @RestMapping 사용하고, 변수명 바꿈(오타수정등)

* refactor :쿼리 파라미터를 @ModelAttribute 사용해서 한버ㅓㄴ에 받도록

* style: 필요없는 거 지움

* chore : application.properties 에 jdbc 설정 추가

* style : toString 삭제

* feat : JDBC를 사용한 레파지토리 Dao 추가

* feat : JDBC Dao 를 사용해서 DB 기능 추가

* style : MapToProductTransformer -> ProductTransformer클래스 명 변경

* docs : README.md 작성

---------

Co-authored-by: 박재성(Jason) <wotjd243@naver.com>

jinseohyun1228 committed:
docs(README.md): 기능 구현 사항 작성

jinseohyun1228 committed:
chore(controller) : 컨트롤러 명 변경

jinseohyun1228 committed:
docs(README.md): 유효성 검사 위치 추가 고민 작성

jinseohyun1228 committed:
refactor(ProductDao) : id 전략 위치 Dao로 변경

jinseohyun1228 committed:
refactor(ProductDao) : id 전략 위치 Dao로 변경

jinseohyun1228 committed:
refactor(ProductDao) : idㄹ를 Dao 내부에서 생성하도록

jinseohyun1228 committed:
build(build.gradle): validation 의존성 추가

jinseohyun1228 committed:
refactor: 상품 추가 로직에 유효성 검사를 추가하고,컨트롤러,서비스 단으로 분리하였음

jinseohyun1228 committed:
refactor: 삼품 삳제 로직에, Dao에서 유효성을 검사할 수 있게, 서비스와 컨트롤러를 구분하도록 수정

jinseohyun1228 committed:
refactor:상품 정보 변경 로직 수정

jinseohyun1228 committed:
fix(

jinseohyun1228 committed:
refactor:상품 정보 찾기  로직 수정, 컨트롤러와 서비스 분리

jinseohyun1228 committed:
fix (ProductService) : Servide내에서 Dao 클래스 주입안돼서 코드 수정

jinseohyun1228 committed:
docs(READE.md): 예외 발생 처리 로직 추가

jinseohyun1228 committed:
docs(READE.md): 예외 클래스 작성

jinseohyun1228 committed:
docs(product.html) : 헤더 추가^^

jinseohyun1228 committed:
docs(error.html) : 스타일 변경

jinseohyun1228 committed:
feat: 에러 핸들러 추가

jinseohyun1228 committed:
bug(ProductValidator): 가격 유효성 검사 로직 추가

jinseohyun1228 committed:
bug(ProductValidator): 가격 유효성 검사 로직 추가

jinseohyun1228 committed:
docs : 타임리프 탬플릿 변경

jinseohyun1228 committed:
fetch: Valid 구현

jinseohyun1228 committed:
fetch: vaild 사용

jinseohyun1228 committed:
build: 의존성 추가

jinseohyun1228 committed:
docs: 구현 고민 정리

jinseohyun1228 committed:
style:

jinseohyun1228 committed:
bug: 유효성 검사 안되는 오류 해결

jinseohyun1228 committed:
Merge branch 'jinseohyun1228' into step1

jinseohyun1228 committed:
stlye: 패키지 구조 변경

jinseohyun1228 committed:
stlye: 패키지 구조 변경

jinseohyun1228 committed:
refactor: 타임리프 -> ajax로!! 로 변환

jinseohyun1228 committed:
refactor: 유효성 검사를 컨트롤러에서 진행

jinseohyun1228 committed:
refactor: DB id 생성 전략으로 변경

jinseohyun1228 committed:
Merge step2_ajax into step1 with preference to step2_ajax changes

jinseohyun1228 committed:
chore: 패키지 변경

jinseohyun1228 committed:
feat : User 객체 구현과 레파지토리 구현

jinseohyun1228 committed:
feat : Jwt 발급 검증 util 객체 구현

jinseohyun1228 committed:
feat : Jwt 발급 검증 util 객체 구현

jinseohyun1228 committed:
feat : 회원가입 로직 구현

jinseohyun1228 committed:
feat :로그인  로직 구현

jinseohyun1228 committed:
refactor:메서드 명 수정

jinseohyun1228 committed:
fix: url 잘못 등록되어있단 부분 고침 -> 무한 리디렉션..

jinseohyun1228 committed:
fix: 쉬운 키로, 생성안되는 오류 해결

jinseohyun1228 committed:
fix: 바디데이터 응답 객체에 담기

jinseohyun1228 committed:
chore(UserException) : 패키지 위치 변경

jinseohyun1228 committed:
fix(UserJoinRequest): 알고보니까 member 페이지에서 키로 잘 전달 중임

jinseohyun1228 committed:
fix(ProductDao): id 발급 전략에 따른 변경

jinseohyun1228 committed:
fix(MemberService) : != 연산자 고침

jinseohyun1228 committed:
fix : 에러 처리 부분

jinseohyun1228 committed:
fix:jwt 검증 부분 오류 해결

jinseohyun1228 committed:
fetch: 커스텀예외 구현

jinseohyun1228 committed:
refactor:오류 핸들링 동일하게 변경

jinseohyun1228 committed:
build: 의존성 추가

jinseohyun1228 committed:
chore: ㄱ공백 추가..?

jinseohyun1228 committed:
docs: 문서 작성

jinseohyun1228 committed:
3단계 과제 요구사항에 맞춰서, 인증 토큰 키 변경

jinseohyun1228 committed:
docs : 기능 구현 계획

jinseohyun1228 committed:
build : 의존성 추가

jinseohyun1228 committed:
refactor: 위시리스트 등록시, 유저 id 가 필요해서 JWT에 id도 추가함(DB 접근 ㄴㄴ)

jinseohyun1228 committed:
fetch : 위시리스트 상품 Dto, Entity, Dao 구현

jinseohyun1228 committed:
refactor: jwt 발급시 id도 추가

jinseohyun1228 committed:
refactor: jwt 발급시 id도 추가

jinseohyun1228 committed:
chore :

jinseohyun1228 committed:
chore : 수정사항 딱히 없음

jinseohyun1228 committed:
refactor: 보안을 위해 password를 뺌

jinseohyun1228 committed:
refactor: 로그인, 회원가입, 유효성 검사 추가

jinseohyun1228 committed:
부산대 BE 진서현 2주차 step1 (#198)

* Initial commit

* feat: set up the project

* Docs : 파일 형식자 변경

* Test : 컨트롤러 잘 돌아가는지..ㅎ

* Feat: 컨트롤러 작성

* Feat: 컨버터 작성

* Feat: Product 엔티티

* Feat: dto 작성

* Feat: 컨트롤러 작성

* fix : getter 없던 오류 추가

* feat: 요청 객체 추가

* fix : 1. 메서드 접근 제어자 수정, 2. id 발급 전략 선택 3.

* fix : 1. 메서드 접근 제어자 수정, 2. id 발급 전략 선택 3. 쿼리파라미터 타입형 long으로 수정 4. 불필요한 예외처리 삭제

* Feat: thymeleaf 공부

* style : ProductRequest의 줄바꿈

* Feat: Product에 toString 추가

* Feat: 타임리프를 적용한 html 문서 추가

* Feat:수정을 위한 별도의 페이지 추가

* Feat: 관리자페이지 관련 컨트롤러 추가

* Feat: 폼에서 put 메서드를 쓰기 위한 빈등록

* chore : 타임리프 사용을 위한 의존성 추가

* Style : 페이지 한글로 통일

* Style : 페이지 한글로 통일

* refactor : 전체 메서드의 매핑 url에 처음을  rest로 수정

* refactor : @RestMapping 사용하고, 변수명 바꿈(오타수정등)

* refactor :쿼리 파라미터를 @ModelAttribute 사용해서 한버ㅓㄴ에 받도록

* style: 필요없는 거 지움

* chore : application.properties 에 jdbc 설정 추가

* style : toString 삭제

* feat : JDBC를 사용한 레파지토리 Dao 추가

* feat : JDBC Dao 를 사용해서 DB 기능 추가

* style : MapToProductTransformer -> ProductTransformer클래스 명 변경

* docs : README.md 작성

* docs(README.md): 기능 구현 사항 작성

* chore(controller) : 컨트롤러 명 변경

* docs(README.md): 유효성 검사 위치 추가 고민 작성

* refactor(ProductDao) : id 전략 위치 Dao로 변경

* refactor(ProductDao) : id 전략 위치 Dao로 변경

* refactor(ProductDao) : idㄹ를 Dao 내부에서 생성하도록

* build(build.gradle): validation 의존성 추가

* refactor: 상품 추가 로직에 유효성 검사를 추가하고,컨트롤러,서비스 단으로 분리하였음

* refactor: 삼품 삳제 로직에, Dao에서 유효성을 검사할 수 있게, 서비스와 컨트롤러를 구분하도록 수정

* refactor:상품 정보 변경 로직 수정

* fix(

* refactor:상품 정보 찾기  로직 수정, 컨트롤러와 서비스 분리

* fix (ProductService) : Servide내에서 Dao 클래스 주입안돼서 코드 수정

* docs(READE.md): 예외 발생 처리 로직 추가

* docs(READE.md): 예외 클래스 작성

* docs(product.html) : 헤더 추가^^

* docs(error.html) : 스타일 변경

* feat: 에러 핸들러 추가

* bug(ProductValidator): 가격 유효성 검사 로직 추가

* bug(ProductValidator): 가격 유효성 검사 로직 추가

* docs : 타임리프 탬플릿 변경

* fetch: Valid 구현

* fetch: vaild 사용

* build: 의존성 추가

* docs: 구현 고민 정리

* style:

* bug: 유효성 검사 안되는 오류 해결

* stlye: 패키지 구조 변경

* stlye: 패키지 구조 변경

* refactor: 타임리프 -> ajax로!! 로 변환

* refactor: 유효성 검사를 컨트롤러에서 진행

* refactor: DB id 생성 전략으로 변경

* Merge step2_ajax into step1 with preference to step2_ajax changes

---------

Co-authored-by: 박재성(Jason) <wotjd243@naver.com>

jinseohyun1228 committed:
부산대 BE_진서현_2주차 과제(2단계) (#332)

* Initial commit

* feat: set up the project

* Docs : 파일 형식자 변경

* Test : 컨트롤러 잘 돌아가는지..ㅎ

* Feat: 컨트롤러 작성

* Feat: 컨버터 작성

* Feat: Product 엔티티

* Feat: dto 작성

* Feat: 컨트롤러 작성

* fix : getter 없던 오류 추가

* feat: 요청 객체 추가

* fix : 1. 메서드 접근 제어자 수정, 2. id 발급 전략 선택 3.

* fix : 1. 메서드 접근 제어자 수정, 2. id 발급 전략 선택 3. 쿼리파라미터 타입형 long으로 수정 4. 불필요한 예외처리 삭제

* Feat: thymeleaf 공부

* style : ProductRequest의 줄바꿈

* Feat: Product에 toString 추가

* Feat: 타임리프를 적용한 html 문서 추가

* Feat:수정을 위한 별도의 페이지 추가

* Feat: 관리자페이지 관련 컨트롤러 추가

* Feat: 폼에서 put 메서드를 쓰기 위한 빈등록

* chore : 타임리프 사용을 위한 의존성 추가

* Style : 페이지 한글로 통일

* Style : 페이지 한글로 통일

* refactor : 전체 메서드의 매핑 url에 처음을  rest로 수정

* refactor : @RestMapping 사용하고, 변수명 바꿈(오타수정등)

* refactor :쿼리 파라미터를 @ModelAttribute 사용해서 한버ㅓㄴ에 받도록

* style: 필요없는 거 지움

* chore : application.properties 에 jdbc 설정 추가

* style : toString 삭제

* feat : JDBC를 사용한 레파지토리 Dao 추가

* feat : JDBC Dao 를 사용해서 DB 기능 추가

* style : MapToProductTransformer -> ProductTransformer클래스 명 변경

* docs : README.md 작성

* docs(README.md): 기능 구현 사항 작성

* chore(controller) : 컨트롤러 명 변경

* docs(README.md): 유효성 검사 위치 추가 고민 작성

* refactor(ProductDao) : id 전략 위치 Dao로 변경

* refactor(ProductDao) : id 전략 위치 Dao로 변경

* refactor(ProductDao) : idㄹ를 Dao 내부에서 생성하도록

* build(build.gradle): validation 의존성 추가

* refactor: 상품 추가 로직에 유효성 검사를 추가하고,컨트롤러,서비스 단으로 분리하였음

* refactor: 삼품 삳제 로직에, Dao에서 유효성을 검사할 수 있게, 서비스와 컨트롤러를 구분하도록 수정

* refactor:상품 정보 변경 로직 수정

* fix(

* refactor:상품 정보 찾기  로직 수정, 컨트롤러와 서비스 분리

* fix (ProductService) : Servide내에서 Dao 클래스 주입안돼서 코드 수정

* docs(READE.md): 예외 발생 처리 로직 추가

* docs(READE.md): 예외 클래스 작성

* docs(product.html) : 헤더 추가^^

* docs(error.html) : 스타일 변경

* feat: 에러 핸들러 추가

* bug(ProductValidator): 가격 유효성 검사 로직 추가

* bug(ProductValidator): 가격 유효성 검사 로직 추가

* docs : 타임리프 탬플릿 변경

* fetch: Valid 구현

* fetch: vaild 사용

* build: 의존성 추가

* docs: 구현 고민 정리

* style:

* bug: 유효성 검사 안되는 오류 해결

* stlye: 패키지 구조 변경

* stlye: 패키지 구조 변경

* refactor: 타임리프 -> ajax로!! 로 변환

* refactor: 유효성 검사를 컨트롤러에서 진행

* refactor: DB id 생성 전략으로 변경

* Merge step2_ajax into step1 with preference to step2_ajax changes

* chore: 패키지 변경

* feat : User 객체 구현과 레파지토리 구현

* feat : Jwt 발급 검증 util 객체 구현

* feat : Jwt 발급 검증 util 객체 구현

* feat : 회원가입 로직 구현

* feat :로그인  로직 구현

* refactor:메서드 명 수정

* fix: url 잘못 등록되어있단 부분 고침 -> 무한 리디렉션..

* fix: 쉬운 키로, 생성안되는 오류 해결

* fix: 바디데이터 응답 객체에 담기

* chore(UserException) : 패키지 위치 변경

* fix(UserJoinRequest): 알고보니까 member 페이지에서 키로 잘 전달 중임

* fix(ProductDao): id 발급 전략에 따른 변경

* fix(MemberService) : != 연산자 고침

* fix : 에러 처리 부분

* fix:jwt 검증 부분 오류 해결

* fetch: 커스텀예외 구현

* refactor:오류 핸들링 동일하게 변경

* build: 의존성 추가

* chore: ㄱ공백 추가..?

* docs: 문서 작성

---------

Co-authored-by: 박재성(Jason) <wotjd243@naver.com>

jinseohyun1228 committed:
refactor: 에러 메시지를 enum으로 구현과 적용

jinseohyun1228 committed:
test : 유저 회원가입 로직 테스트 구현

jinseohyun1228 committed:
chore: member -> user로 통일

jinseohyun1228 committed:
chore: member -> user로 통일

jinseohyun1228 committed:
Merge branch 'jinseohyun1228' into step3

jinseohyun1228 committed:
Refactor: URI를 Restful 한 설계에 맞춰변경

jinseohyun1228 committed:
Refactor: URI를 Restful 한 설계에 맞춰변경

jinseohyun1228 committed:
Refactor: jwt 인증에 필요없는 이메일 값을 제외함, enum을 사용해서 에러 메시지를 설정함,         HttpHeaders.AUTHORIZATION

jinseohyun1228 committed:
refator:products 반환 방식을 Map 형태에서 객체 그대로 반환하는 방법으로 변경

jinseohyun1228 committed:
refactor: user의 role enum 적용

jinseohyun1228 committed:
fetch(AuthUtil): 세션의 user가져오는 메서드 추가

jinseohyun1228 committed:
refactor: AuthUtil -> JwtUtil로 클래스 명 변경

jinseohyun1228 committed:
refactor: AuthUtil -> Jwtutil 변경

jinseohyun1228 committed:
style: 패키지 구조 변경

jinseohyun1228 committed:
refactor: session에서 user 가져오기 어노테이션으로 구현

jinseohyun1228 committed:
chore: 위치 변경 등

jinseohyun1228 committed:
refactor: 테이블 생성 위치 스크립트로 변경

jinseohyun1228 committed:
테스트 파일 올라가도록 설정 변경

jinseohyun1228 committed:
충돌 해결

jinseohyun1228 committed:
충돌 해결 후

jinseohyun1228 committed:
docs : 기능 요구사항 정리

jinseohyun1228 committed:
build:JPA 의존성 추가

jinseohyun1228 committed:
fetch: wishProduct 에 JPA 도입

jinseohyun1228 committed:
fetch: User JPA 도입

jinseohyun1228 committed:
refator: product를 JPA로 관리

jinseohyun1228 committed:
fetch: 제품 관련 에러 추가

jinseohyun1228 committed:
chore: 자바 코드컨벤션 지키기

jinseohyun1228 committed:
chore: 자바 코드컨벤션 지키기

jinseohyun1228 committed:
refator: wishProduct Serive로 변경

jinseohyun1228 committed:
fix: User객체 ENUM 적용으로 인한 테스트 코드 오류 수정

jinseohyun1228 committed:
refactor : WishProductRepository 의 deleteByProductIdAndUserId메서드 반환값 수정

jinseohyun1228 committed:
fetch: WishProductRepositoryTest 추가

jinseohyun1228 committed:
fetch: ProductRepositoryTest 추가

jinseohyun1228 committed:
fetch: UserRepositoryTest 추가

jinseohyun1228 committed:
Merge branch 'jinseohyun1228' into step1

jinseohyun1228 committed:
chore:빈칸 , 공백 삭제 등

jinseohyun1228 committed:
Merge branch 'step1' of https://github.com/jinseohyun1228/spring-gift-jpa into step1

jinseohyun1228 committed:
git : 충돌 해결

jinseohyun1228 committed:
git : 충돌 해결

jinseohyun1228 committed:
refator: Product 업데이트 시 setter가 아닌 update메서드를 사용할 수 있도록 변경

jinseohyun1228 committed:
refator: Product 업데이트 시 setter가 아닌 update메서드를 사용할 수 있도록 변경,
+  AdminProductController에서 Admin삭제

jinseohyun1228 committed:
fix : 불필요한 @Transactional삭제

jinseohyun1228 committed:
fetch : 위시물건:  물건-유저 연관관계 등록

jinseohyun1228 committed:
refactor: 레파지토리 반환값을 Optional으로 변경

jinseohyun1228 committed:
refactor(WishProductController): 반환 메시지를 영어로

jinseohyun1228 committed:
fetch : Aop Aspect로 메서드 권한 검사하기

jinseohyun1228 committed:
refactor : Product에 판매자 연관관계 추가

jinseohyun1228 committed:
refactor: 응답을 위한 ProductResponce추가

jinseohyun1228 committed:
refactor: WishProductController에서 getProducts()의 반환값 ProductResponce로 수정

jinseohyun1228 committed:
style: 전체 코드 컨벤션 수정

jinseohyun1228 committed:
fix : 회원가입시 role이 소문자로 들어오는 부분 수정

jinseohyun1228 committed:
fix... : aspect 왜 안될까

jinseohyun1228 committed:
fix: Aspect 도입 고민 중으로 해당 로직 삭제

jinseohyun1228 committed:
docs: readme 파일에 기능 구현 작성

jinseohyun1228 committed:
fetch: 일반 유저의 상품 리스트 페이지와 장바구니 페이지 구현

jinseohyun1228 committed:
refactor: 회원가입 요청 주소를 register에서 join으로 변경
1. WepConfig에서 변경
2. Controller에서 변경
3. html페이지에서 요청 주소 변경

jinseohyun1228 committed:
fix : 관리자 상품 관련 버그 고침
1. html 페이지에서 삭제,수정  요청 주소가 /{id}가 아닌 쿼리 파라미터 형식이라 변경
2. 컨트롤러 단에서 삭제 수정 요청 주소를 쿼리파라미터로 받고 있는 부분 변경
3. html 페이지 내에서 조회 로직 내에서 사용하는 변수명이 달라 생기는 에러 수정

jinseohyun1228 committed:
refactor: 중복 로직이던 상품 리스트 조회 부분 ProductController로 개별 구현
-> 중복 로직 삭제와 html 페이지에서 관리가 편해짐

jinseohyun1228 committed:
fetch(memberWishList.html): 유저용 상품 리스트 조회 페이지 추가

jinseohyun1228 committed:
Revert "fetch: 위시리스트 관련 페이지와 로직 추가

1. 멤버의 위시리스트 페이지와 상품 조회 페이지를 각각 추가
2. 위시리스트 조회에 쓰이는 WishProductResponce 객체 추가와 옵셔널로 묶인 응답 List로 변환 기능 추가

This reverts commit 39034b1d31d79e604e7e37a1d2d8fb16b1833472.

jinseohyun1228 committed:
fetch: 위시리스트 관련 페이지와 로직 추가
1. 멤버의 위시리스트 페이지와 상품 조회 페이지를 각각 추가
2. 위시리스트 조회에 쓰이는 WishProductResponce 객체 추가와 옵셔널로 묶인 응답 List로 변환 기능 추가

jinseohyun1228 committed:
chore: 코드 컨벤션

jinseohyun1228 committed:
커밋이 어디로..?

jinseohyun1228 committed:
fetch : 물건리스트  페이지로 볼 수 있게 변경

[변경 사항]
1. ProductController : Page 객체를 반환하는 getProductPage 메서드 추가
2. ProductService:  Page 객체를 반환하는 getProductPage 메서드 추가
3. ProductRepository : Page 객체를반환할 수 있도록 메서드 추가
4. memberProduct.html : get 요청 변경

---

jinseohyun1228 committed:
fetch : 물건리스트  페이지로 볼 수 있게 변경

[변경 사항]
1. ProductController : Page 객체를 반환하는 getProductPage 메서드 추가
2. ProductService:  Page 객체를 반환하는 getProductPage 메서드 추가
3. ProductRepository : Page 객체를반환할 수 있도록 메서드 추가
4. memberProduct.html : get 요청 변경

---

jinseohyun1228 committed:
fix: 위시리스트에 상품이 중복으로 담기는 문제 해결

jinseohyun1228 committed:
fix: 위시리스트에 상품이 중복으로 담기는 문제 해결

jinseohyun1228 committed:
Revert "fix: 위시리스트에 상품이 중복으로 담기는 문제 해결"

This reverts commit 0c1e4a2eb1c16cf4144d856f349dc7594025b110.

jinseohyun1228 committed:
Revert "fix: 위시리스트에 상품이 중복으로 담기는 문제 해결"

This reverts commit 2ad0946a2621da1a80215e7109f0888af1b51b90.

jinseohyun1228 committed:
Revert "fetch : 물건리스트  페이지로 볼 수 있게 변경"

This reverts commit 4893d9919997f2c40056b3d51c8672d99401a754.

jinseohyun1228 committed:
Revert "fetch : 물건리스트  페이지로 볼 수 있게 변경"

This reverts commit 9368a25e744fe8eae15b3509e1f15f926fe863f3.

jinseohyun1228 committed:
fix : 장바구니 담기가 중복으로 되는 문제 해결

jinseohyun1228 committed:
fetch : 물건리스트  페이지로 볼 수 있게 변경

[변경 사항]
1. ProductController : Page 객체를 반환하는 getProductPage 메서드 추가
2. ProductService:  Page 객체를 반환하는 getProductPage 메서드 추가
3. ProductRepository : Page 객체를반환할 수 있도록 메서드 추가
4. memberProduct.html : get 요청 변경

---

jinseohyun1228 committed:
fix : 장바구니 담기가 중복으로 되는 문제 해결

jinseohyun1228 committed:
fetch : 물건리스트  페이지로 볼 수 있게 변경

[변경 사항]
1. ProductController : Page 객체를 반환하는 getProductPage 메서드 추가
2. ProductService:  Page 객체를 반환하는 getProductPage 메서드 추가
3. ProductRepository : Page 객체를반환할 수 있도록 메서드 추가
4. memberProduct.html : get 요청 변경

---

jinseohyun1228 committed:
fetch: 상품 위시리스트를 페이지로 제공

1. memberWishList.html : 페이지를 받을 수 있게 문서 수정
2. WishProductRepository: page 객체를 반환하는 메서드 추가
3. WishProductService: 메서드 추가
4. WishlistController : 메서드 추가

jinseohyun1228 committed:
fetch(adminProduct.html) : 관리페이지에서도 물건리스트를 페이지로 볼 수 있게 수정

jinseohyun1228 committed:
style : 페이지에 헤더 추가

jinseohyun1228 committed:
chore: 전체적인 코드 컨벤션

jinseohyun1228 committed:
refactor : 이후에 페이지에서  한번에 출력할 개수를 지정할 수 있도롤 파라미터 추가

1. wishProductContoller과 Service 수정
2. productContoller과 Service 수정

jinseohyun1228 committed:
Merging with upstream/main using ours strategy

hocaron commented on src/main/java/gift/main/controller/ProductController.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    @GetMapping("/productpage/{pageNum}")
+    public ResponseEntity<?> getProductPage(@PathVariable(name = "pageNum") Optional<Integer> pageNum,@PathVariable(name = "productNum") Optional<Integer> productNum) {
```
Controller 에서 Pageable 로 받을 수도 있을 것 같아요😀 sorting 도 지원해줘서 간편하게 사용할 수 있습니다.

jinseohyun1228 replied:
해당 부분 반영했습니다!
페이지에서도 size를 사용자가 설정할 수 있도록 기능 추가했습니다 😄😄 
입력으로 받으니까 간편한것같습니다! 감사합니다1!

hocaron commented on src/main/java/gift/main/config/WebConfig.java:
```diff
@@ -14,11 +13,11 @@
 @Configuration
 @EnableWebMvc
-public class WebConfig  implements WebMvcConfigurer {
+@EnableAspectJAutoProxy(proxyTargetClass = true)
```
해당 설정없이도 AOP 동작이 가능할 것 같아요😀

hocaron reviewed (approved):
서현님, 안녕하세요👋

페이징 적용해주셨네요👍
함께 작업하는 입장에서 궁금한 점이 생겨 코멘트 드렸습니다. 확인 부탁드릴게요.

3주차 고생 많으셨습니다🙇‍♀️

hocaron commented:
2단계 코멘트 남긴 부분 중에 Role 에 관해 Int 가 아닌 다른 값으로 대체해보면 좋을 것 같다는 내용이 있었는데, 해당 내용도 확인 부탁드릴게요.

jinseohyun1228 committed:
test: enum과 string을 equals으로 비교시 결과 테스트

jinseohyun1228 committed:
fix:UserEntity의 role칼럼이 Sting 형태로 DB에 저장되도록 수정

jinseohyun1228 committed:
refactor(User관련 클래스) : Role 내부 필드를 enum타입 Role로 변경

[변경 위치]
1. UserVo
2. UserJoinRequest
3. UserDto

[변경 이유]
User Entity의 Role 필드가 Enum으로 되어있어 예상 가능한 동작을 할 수 잇도록 나머지 객체 변경,
String 타입보다 enum 이 안정성을 더 보장할 수 있다 판단 됨.

jinseohyun1228 committed:
fetch :  UserJoinRequest의 필드가 enum이 되면서 소문자 입력을 받지 못하는 부분 커스텀 Deserializer 도입

jinseohyun1228 committed:
refactor: 페이지네이션에서 pageable을 컨트롤러 입력 파라미터로 받기 + 안쓰는 메서드 삭제

jinseohyun1228 committed:
refactor: 페이지네이션에서 pageable을 컨트롤러 입력 파라미터로 받기 + 안쓰는 메서드 삭제

jinseohyun1228 committed:
fix:User의 Role 칼럼 값이 TINYINT으로 올라가는 부분 수정

jinseohyun1228 commented:
> 현재 User 테이블의 Role 컬럼 타입이 TINYINT 값(enum 순서)로 저장되는 것 같아요.
같이 작업하는 개발자가 이 히스토리를 모르고 enum 순서를 수정한다면, 다른 권한으로 체크가 될 것 같은데, 어떻게 방지할 수 있을까요?!😀

해당 부분 `@Enumerated(EnumType.STRING)`사용해서 변경하였습니다.
자동 매핑이 된다고 생각했는데,,, 이 부분 공부하면서 다른 타입의 어노테이션도 있다는 것을 알게되었습니다! 리뷰 감사합니다  🙇

jinseohyun1228 commented:
> 현재 User 테이블의 Role 컬럼 타입이 TINYINT 값(enum 순서)로 저장되는 것 같아요.
같이 작업하는 개발자가 이 히스토리를 모르고 enum 순서를 수정한다면, 다른 권한으로 체크가 될 것 같은데, 어떻게 방지할 수 있을까요?!😀

해당 부분 `@Enumerated(EnumType.STRING)`사용해서 변경하였습니다.
자동 매핑이 된다고 생각했는데,,, 이 부분 공부하면서 다른 타입의 어노테이션도 있다는 것을 알게되었습니다! 리뷰 감사합니다  🙇


> userVo 에서 가져온 getRole 은 타입이 String 이라 해당 조건문은 항상 true 가 될 것 같아요! 어떻게 변경할 수 있을까요?!

혹시 이 부분 항상 false가 된다고 말씀하시려고 하신것 맞나용..??

### 단순 해결 방법 : 문제 발생 지점 변경

만약 UserVo의 role을 String으로 유지해야한다면, Role값을 문자열 반환 메서드를 사욜할 수 있을 것 같습니다.

```
public String getRole() {
        return role;
    }

<사용 코드>
         Role.ADMIN.getRole().equals(user.getRole()) 

```

### 코드 자체에 문제가 있다고 느낌...
하지만 저도 User의 Role값이 enum일 것이라고 생각되어 처음과 같은 코드를 짠 것 같습니다.

이 부분 이후 문제가 발생될 수 있다고 생각되고, 예상되는 기능(User의 Role값이 enum일 것)과 달라 좋은 코드가 아니라고 생각했습니다.

### User관련 클래스의 Role 필드 : String --> enum 
**따라서 User관련 클래스의 Role 필드를 모두 String에서 enum 타입으로 변경하였습니다.**

### 새로운 문제 발생
이렇게 변경하고 나니 컨트롤러 요청  UserJoinRequest객체에서 역직렬화 시 소문자의 경우 예외가 발생하였습니다.


###  해결 방법1. 커스텀 Deserializer 사용해서 역직렬시 에러가 나지 않도록 구현
이 부분은 검색을 통해 커스텀 Deserializer 사용해서 역직렬시 에러가 나지 않도록 구현하였습니다!!

1. `JsonDeserializer<?>` 클래스 구현
2. `deserialize` 으로 로직 설정
    
    ```java
    public class ConverterRoleDeserializer extends JsonDeserializer<Role> {
        @Override
        public Role deserialize(JsonParser p, DeserializationContext ctxt) throws IOException, JacksonException {
            return Role.toRole(p.getValueAsString().toUpperCase());
        }
    }
    ```
    
3. 해당 역직렬로직을 사용하고 싶은 필드에 `@JsonDeserialize` + `속성 : using`으로 설정
    
    ```java
            @JsonDeserialize(using = ConverterRoleDeserializer.class)
            Role role) {
    ```

jinseohyun1228 commented:
그런데 이 과정에서 궁금한 부분이 두가지 생겼습니다!

### User 관련 클래스에 String role을 받는 생성자를 두어야할까요?  
UserJoinRequest의 내부 필드를 enum으로 바꾸기 전에는 role값이 String으로 들어오는 경우가 발생하기 때문에, 어느 시점에 String role값이 살아있을지(?) 몰라  여기저기에 String으로 role값을 받을때 생성할 수 있는 생성자를 두었습니다.

그러나 현재 입력 자체를 Role값으로 받을 수 있어 문자열 role값이 들어오지 않습니다.
이런경우 해당 생성자를 두는 것이 좋울지 (이후에 사용될 수도 있으니..) 혹은 바로 삭제하는 것이 좋을지 궁금합니다.

### 좋은 코드가 무엇인지, 특히 짧은 코드가 좋다는 의견에 어떻게 생각하시는지 궁금합니다.
이번 코드 수정에 단순 해결 방법을 적용할 수도 있었지만, 제가 생각하는 좋은 코드에 위배 되어서 변경하였습니다.

제가 생각하는 좋은 코드란 다음과 같습니다.

< 지금까지 생각했을 때 좋은 코드>
> 
> **1순위: 빠르게 요구 사항을 구현한 코드**> 
>
> **2순위: 예상한대로 동작하는 코드**
> - 나조차 기억을 못하면 다른 동료는 얼마나 난처할지 고민해보자!
> 
> **3순위: 그 다음이 짧은 코드…**

짧은 코드가 좋다고 느낀데에는 _먼저 짧은 코드를 지키자!!_라는 이야기를 여러 글에서 보았고,,, 일단 따라보자!라는 결론에 직접 과제를 할때 짧은 코드를 작성하려고 노력했습니다.

실제로 짧은 코드는 
1) 읽는 입장에서 부담이 줄어들었고
 2) 짧은 코드를 작성하기 위해서라도 메서드 역할이 줄어들게 되었습니다.


짧은 코드를 지향하는 이유가 짧은코드 그 자체인지, 메서드의 복잡도를 낮추기 위함인지 궁금합니다. 
강의시간에도 클린코드를 짜는 방법으로 짧은 메서드길이를 유지하라는 내용이 있었습니다.    


실제 현업에서도 짧은 코드가 많이 중요한가요?? 또한 짧은 코드보다 우선시 되는 것이 있다면 무엇일지 궁금합니다!

jinseohyun1228 commented:
그런데 이 과정에서 궁금한 부분이 두가지 생겼습니다!

### User 관련 클래스에 String role을 받는 생성자를 두어야할까요?  
UserJoinRequest의 내부 필드를 enum으로 바꾸기 전에는 role값이 String으로 들어오는 경우가 발생하기 때문에, 어느 시점에 String role값이 살아있을지(?) 몰라  여기저기에 String으로 role값을 받을때 생성할 수 있는 생성자를 두었습니다.

그러나 현재 입력 자체를 Role값으로 받을 수 있어 문자열 role값이 들어오지 않습니다.
이런경우 해당 생성자를 두는 것이 좋울지 (이후에 사용될 수도 있으니..) 혹은 바로 삭제하는 것이 좋을지 궁금합니다.

### 좋은 코드가 무엇인지, 특히 짧은 코드가 좋다는 의견에 어떻게 생각하시는지 궁금합니다.
이번 코드 수정에 단순 해결 방법을 적용할 수도 있었지만, 제가 생각하는 좋은 코드에 위배 되어서 변경하였습니다.

제가 생각하는 좋은 코드란 다음과 같습니다.

< 지금까지 생각했을 때 좋은 코드>
> 
> **1순위: 빠르게 요구 사항을 구현한 코드**> 
>
> **2순위: 예상한대로 동작하는 코드**
> - 나조차 기억을 못하면 다른 동료는 얼마나 난처할지 고민해보자!
> 
> **3순위: 그 다음이 짧은 코드…**

짧은 코드가 좋다고 느낀데에는 _먼저 짧은 코드를 지키자!!_라는 이야기를 여러 글에서 보았고,,, 일단 따라보자!라는 결론에 직접 과제를 할때 짧은 코드를 작성하려고 노력했습니다.

실제로 짧은 코드는 
1) 읽는 입장에서 부담이 줄어들었고
 2) 짧은 코드를 작성하기 위해서라도 메서드 역할이 줄어들게 되었습니다.


짧은 코드를 지향하는 이유가 짧은코드 그 자체인지, 메서드의 복잡도를 낮추기 위함인지 궁금합니다. 
강의시간에도 클린코드를 짜는 방법으로 짧은 메서드길이를 유지하라는 내용이 있었습니다.    


실제 현업에서도 짧은 코드가 많이 중요한가요?? 또한 짧은 코드보다 우선시 되는 것이 있다면 무엇일지 궁금합니다!

jinseohyun1228 committed:
fix:UserVo 의 role -> enum 변경으로 getRole이 enum 반환하도록 변경

jinseohyun1228 commented:
Aspect은... 알려주신대로 해봤는데 잘 안되는 것 같아서 더 공부중입니다..!! 감사합니다.

hocaron commented:
@jinseohyun1228 서현님, 안녕하세요.

1. role 의 경우, enum 타입으로 전달 받는 것이 적절해보입니다! requestDto 에서 정의 되지 않은 경우도 validation 할 수 있어서, 비지니스 로직에서는 정의된 role 만을 다룰 수 있어 좋을 것 같아요.

요청 필드에 role(소문자) 필드로 요청해도 될 것 같은데, 대문자로 변경하신 이유가 있을까요?!! 클라이언트에서 보내는 요청값인데, 해당 필드만 대문자인 부분이 궁금해서 여쭤봅니다. 소문자로 해도 된다면 커스텀 Deserializer 도 필요없을 것 같아서요.

```json
curl --location 'http://localhost:8080/spring-gift/members/join' \
--header 'Content-Type: application/json' \
--data-raw '{
    "name" : "테스트",
    "email" : "test@emai.com",
    "password" : "testPassword",
    "role" : "ADMIN"
}'
```

hocaron commented:
1. 좋은 코드가 무엇인지, 특히 짧은 코드가 좋다는 의견에 어떻게 생각하시는지 궁금합니다.
이번 코드 수정에 단순 해결 방법을 적용할 수도 있었지만, 제가 생각하는 좋은 코드에 위배 되어서 변경하였습니다.

제일 좋은 코드는 읽기 좋은 코드라고 생각해요. stream 체이닝으로 조금 더 짧은 코드로 표현할 수 있는 케이스에서 일부러 If 문 과 의도적인 띄어쓰기를 통해서 복잡한 비지니스 로직이 있는 코드를 단락단락 나타내려고 노력합니다. 반대로 너무 긴 코드의 경우는 그 안에서 분리할 수 있는 메서드를 찾아 분리하려고 합니다!

jinseohyun1228 commented:
> @jinseohyun1228 서현님, 안녕하세요.
> 
> 1. role 의 경우, enum 타입으로 전달 받는 것이 적절해보입니다! requestDto 에서 정의 되지 않은 경우도 validation 할 수 있어서, 비지니스 로직에서는 정의된 role 만을 다룰 수 있어 좋을 것 같아요.
> 
> 요청 필드에 role(소문자) 필드로 요청해도 될 것 같은데, 대문자로 변경하신 이유가 있을까요?!! 클라이언트에서 보내는 요청값인데, 해당 필드만 대문자인 부분이 궁금해서 여쭤봅니다. 소문자로 해도 된다면 커스텀 Deserializer 도 필요없을 것 같아서요.
> 
> ```json
> curl --location 'http://localhost:8080/spring-gift/members/join' \
> --header 'Content-Type: application/json' \
> --data-raw '{
>     "name" : "테스트",
>     "email" : "test@emai.com",
>     "password" : "testPassword",
>     "role" : "ADMIN"
> }'
> ```

혹시 지금 enum으로 전달받고 있는 상황이 아닌가요?? User관련 클래스들은 모두 role로 변경했는데 혹시 다른 곳에서 발생하는 문제인가 싶어 여쭤봅니다!!

제 Role enum이 대문자 USER,ADMIN으로 되어 있어서 소문자로 받으니까 오류가 나는 상황이 발생하더라구요..!
Json 파싱은 대소문자를 구분해서  enum 문자열과 동일한 것만 넣어줘 오류가 나는 것으로 이해했습니다!!

혹시 제가 잘못알고 있는 부분일까요?

hocaron commented:
아하! enum 필드를 소문자로 입력받는 경우에 예외가 발생하여 커스텀 Deserializer 를 구현하신 부분이라고 이해했습니다!

클라이언트에서 해당 필드를 보낼 때, 대문자로 보내도록 서로 인터페이스를 맞추면 될 것 같습니다😀 이 부분은 클라이언트에게 자유롭게 제공해주지 않아도 된다고 생각하는데 서현님은 어떻게 생각하실까욧.??!!

hocaron commented:
AOP 의 경우는, userVo 가 어디 위치해있는가에 따라 실행 동작 여부가 달라질 수 있을 것 같아요😀
이런 히스토리를 모르는 개발자가 리팩토링하다가 매개변수 순서를 바꿔버리는 경우에 aop 가 동작하지 않는 이슈가 발생할 수 있어, 이 부분을 고려하고 코드 작성해보면 좋을 것 같습니다!

userVo 가 메서드 매개변수중 제일 앞에 있는 경우
```java
@Before("execution(* gift.main.controller.ProductController.*(..)) && args(userVo, ..)")
```

userVo 가 메서드 매개변수중 제일 뒤에 있는 경우
```java
@Before("execution(* gift.main.controller.ProductController.*(..)) && args(.., userVo)")
```

jinseohyun1228 commented:
>아하! enum 필드를 소문자로 입력받는 경우에 예외가 발생하여 커스텀 Deserializer 를 구현하신 부분이라고 이해했습니다!
>
>클라이언트에서 해당 필드를 보낼 때, 대문자로 보내도록 서로 인터페이스를 맞추면 될 것 같습니다😀 이 부분은 클라이언트에게 자유롭게 제공해주지 않아도 된다고 생각하는데 서현님은 어떻게 생각하실까욧.??!!

앗! 이정도는 프론트단이나 클라이언트에게 제약을 줄 수 있는 부분일까요?
권합 입력을  직접 입력(타이핑)등으로 받지는 않을 것 같지만, 프론트의 소소한 실수나, 대문자 소문자정도의 차이는 백엔드에서 유도리(?)있게 다룰 수 있으면 좋지 않나라는 생각이 들었습니다.  또한 은연중에 이용 유저인 제가 직접 타이핑을 하는 상황이라,  실수로 소문자를 넣을 수도 있겠다는 생각이 들어 해당 기능을 추가한 것 같습니다. 

실무에서는 이런 요청과 응답객체의 조건들이 존재하기 때문에 프론트단에서도 이를 지켜주시겠죠?

hocaron commented:
이름 처럼 자유로운 값이 아닌, ADMIN / USER 등으로 정해져있는 값이기 때문에, 클라이언트 구현도 사용자로부터 직접 권한을 입력 받는 방식이 아닌 드랍다운 형식으로 구현이 될 것 같아요. 

swift, kotlin, javascript 등을 사용하는 클라이언트도 해당 값들을 enum 화(또는 상수화)하여 다루기 때문에, 서버쪽에서 추가적인 처리를 하지 않아도 될 것 같습니다😀

jinseohyun1228 commented:
>이름 처럼 자유로운 값이 아닌, ADMIN / USER 등으로 정해져있는 값이기 때문에, 클라이언트 구현도 사용자로부터 직접 권한을 입력 받는 방식이 아닌 드랍다운 형식으로 구현이 될 것 같아요.
>
>swift, kotlin, javascript 등을 사용하는 클라이언트도 해당 값들을 enum 화(또는 상수화)하여 다루기 때문에, 서버쪽에서 추가적인 처리를 하지 않아도 될 것 같습니다😀

아하 그렇군요!! 감사합니다 해당 내용 알게되었습니다!! 덕분에 실제 개발이 어떻게 이루어지는지 알게되었습니다~! 감사합니다!!

hocaron commented:
서현님 해당 pr 은 머지하겠습니다🫡

고생 많으셨습니다! 이번주차도 같이 화이팅해요👍

jinseohyun1228 commented:
>제일 좋은 코드는 읽기 좋은 코드라고 생각해요. stream 체이닝으로 조금 더 짧은 코드로 표현할 수 있는 케이스에서 일부러 If 문 과 의도적인 띄어쓰기를 통해서 복잡한 비지니스 로직이 있는 코드를 단락단락 나타내려고 노력합니다. 반대로 너무 긴 코드의 경우는 그 안에서 분리할 수 있는 메서드를 찾아 분리하려고 합니다!

메서드 체이닝 기법을 사용하면, 코드가 짧아져 좋은 문법이라고 생각했는데 가독성을 저하시킨다는 문제점이 있을 것 같습니다. 

관련해서 팀원의 코드중  get메서드가 반복되는 코드에 대해 다른 멘토님이 리뷰를 남겨주셨는데, 지금 생각해보니 가독성 저하가 핵심이었던 것 같습니다. 

제가 생각했던 @valid는 가독성이 낮아진다고 생각했지만 오히려 사용빈도가 높아 해당문제가 발생하지 않았던 점을 생각하면, 코드마다 생각해봐야할 점이 많은 것 같습니다.

답변감사합니다!!

jinseohyun1228 commented:
> AOP 의 경우는, userVo 가 어디 위치해있는가에 따라 실행 동작 여부가 달라질 수 있을 것 같아요😀
이런 히스토리를 모르는 개발자가 리팩토링하다가 매개변수 순서를 바꿔버리는 경우에 aop 가 동작하지 않는 이슈가 발생할 수 있어, 이 부분을 고려하고 코드 작성해보면 좋을 것 같습니다!

이부분 확인했습니다! 감사합니다!
